include/alpaka/pltf/PltfCpuSyclIntel.hpp

```diff
@@ -0,0 +1,67 @@
+/* Copyright 2022 Jan Stephan
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#if defined(ALPAKA_ACC_SYCL_ENABLED) && defined(ALPAKA_SYCL_BACKEND_ONEAPI) && defined(ALPAKA_SYCL_ONEAPI_CPU)
+
+#    include <alpaka/dev/DevGenericSycl.hpp>
+#    include <alpaka/dev/Traits.hpp>
+#    include <alpaka/pltf/PltfGenericSycl.hpp>
+
+#    include <CL/sycl.hpp>
+
+#    include <string>
+
+namespace alpaka::experimental
+{
+    namespace detail
+    {
+        // Prevent clang from annoying us with warnings about emitting too many vtables. These are discarded by the
+        // linker anyway.
+#    if BOOST_COMP_CLANG
+#        pragma clang diagnostic push
+#        pragma clang diagnostic ignored "-Wweak-vtables"
+#    endif
+        struct IntelCpuSelector final : sycl::device_selector
+        {
+            auto operator()(sycl::device const& dev) const -> int override
+            {
+                auto const vendor = dev.get_info<sycl::info::device::vendor>();
+                auto const is_intel_cpu = (vendor.find("Intel(R) Corporation") != std::string::npos) && dev.is_cpu();
```

Comment:
does it really not work on any non-Intel x86 cpu ?

Reply:
For now we only target the Intel compiler (and its fork from Xilinx). I don't think they support something else than Intel CPUs at the moment. In the future we do want to expand this to other CPU types as well if they have a SYCL implementation. The Codeplay compiler comes to mind which can target ARM IIRC.

Reply:
I was more thinking about AMD x86 cpus... will the OneAPI cpu backend work on them ?

Reply:
AFAIK it will not. The Intel compiler translates SYCL code into SPIR-V. This is then consumed by one of Intel's OpenCL compilers (CPU, GPU, FPGA) and turned into something device-specific. In theory this should also work with other OpenCL implementations which can digest SPIR-V (as long as no Intel-specific extensions are used) but as far as I'm aware AMD no longer provides an OpenCL runtime for CPUs. And even when they did it had no SPIR-V support.

I don't know whether Intel's CPU OpenCL implementation will work with AMD CPUs.

Reply:
OK, I see.